File: src/server/Modules/People/Modules.People.Core/Entities/Attendance.cs
using FluentPOS.Shared.Core.Domain;
using FluentPOS.Shared.DTOs.Enums;
using System;

namespace FluentPOS.Modules.People.Core.Entities
{
    public class Attendance : BaseEntity
    {
        public long EmployeeId { get; set; }

        public long DepartmentId { get; set; }

        public long PolicyId { get; set; }

        public long DesignationId { get; set; }

        public DateTime AttendanceDate { get; set; }

        public AttendanceStatus AttendanceStatus { get; set; }

        public DateTime AddedOn { get; set; }

        /// <summary>
        /// Bio Or Request.
        /// </summary>
        public AttendanceType AttendanceType { get; set; }

        public long? RequestId { get; set; }

        public DateTime ActualIn { get; set; }

        public DateTime ActualOut { get; set; }

        public DateTime ExpectedIn { get; set; }

        public DateTime ExpectedOut { get; set; }

        public DateTime? CheckIn { get; set; }

        public DateTime? CheckOut { get; set; }

        public RequestStatus Status { get; set; }

        public DateTime StatusUpdateOn { get; set; }

        public long ApprovedBy { get; set; }

        public string Reason { get; set; }

        public double EarnedHours { get; set; }

        public double EarnedMinutes { get; set; }

        public double OvertimeHours { get; set; }

        public double OvertimeMinutes { get; set; }

        public string BioMachineId { get; set; }

        public double DeductedHours { get; set; }

        public double LateMinutes { get; set; }

        /// <summary>
        /// Only contains shift hours or overtime if seperate entry
        /// </summary>
        public double TotalEarnedHours { get; set; }

        public double ActualEarnedHours { get; set; }

        public bool IsLateComer { get; set; }

        public bool IsCheckOutMissing { get; set; }

        public OverTimeType OverTimeType { get; set; }

        public double Production { get; set; }

        public double RequiredProduction { get; set; }

        public bool IsNextDay { get; set; }

        public string PunchCode { get; set; }

        public string CardNo { get; set; }

        public ShiftType ShiftType { get; set; }

        public long? ShiftId { get; set; }

        public long? OvertimeRequestId { get; set; }
    }
}

File: src/client/src/app/modules/admin/people/models/attendance.ts
import { AttendanceStatus } from "src/app/core/enums/AttendanceStatus";
import { AttendanceType } from "src/app/core/enums/AttendanceType";
import { RequestStatus } from "src/app/core/enums/RequestStatus";

export interface Attendance {
    id: string;

    createaAt: string;

    updatedAt: string;

    organizationId: string;

    branchId: string;

    employeeId: string;

    departmentId: string;

    policyId: string;

    designationId: string;

    attendanceDate: Date;

    addedOn: Date;

    attendanceType: AttendanceType;

    attendanceStatus: AttendanceStatus;

    requestId: string;

    actualIn: Date;

    actualOut: Date;

    expectedIn: Date;

    expectedOut: Date;

    checkIn: Date;

    checkOut: Date;
    checkInTime: Date;

    checkOutTime: Date;
    status: RequestStatus;

    statusUpdateOn: Date;

    approvedBy: string;

    reason: string;

    earnedHours: number;

    earnedMinutes: number;

    overtimeHours: number;

    overtimeMinutes: number;

    bioMachineId: string;

    deductedHours: number;

    lateMinutes: number;

    totalEarnedHours: number;

    actualEarnedHours: number;
    employeeName: string;
    statusName: string;
    attendanceStatusName: string;
    attendanceTypeName: string;
    View: boolean;
    Update: boolean;
    Remove: boolean;
    className?: string;
    isCheckOutMissing: boolean;
    isLateComer: boolean;
    punchCode?: number;
    departmentName?: string;
    isNextDay: boolean;
}
